src/BugsnagUnity/Delivery.cs

```diff
@@ -69,11 +72,108 @@ public void Deliver(IPayload payload)
                 report.ApplyEventPayload();
             }
             try
+            {
+                var serialisedPayload = SerializePayload(payload);
+                if (serialisedPayload.Length > MAX_PAYLOAD_BYTES)
+                {
+                    serialisedPayload = TruncateMetadata(payload);
+                }
+                MainThreadDispatchBehaviour.Instance().Enqueue(PushToServer(payload, serialisedPayload));
+            }
+            catch 
             {
                 // not avaliable in unit tests
-                MainThreadDispatchBehaviour.Instance().Enqueue(PushToServer(payload, SerializePayload(payload)));
             }
-            catch { }
+        }
+
+        private byte[] TruncateMetadata(IPayload payload)
+        {
+            var @event = payload.GetSerialisablePayload()["event"] as PayloadDictionary;
+            var metadata = @event["metaData"] as PayloadDictionary;
+            foreach (var section in metadata)
+            {
+                if (section.Key != "app" && section.Key != "device")
+                {
+                    TruncateStringsInMetadataSection(section.Value as Dictionary<string,object>);
+                }
+            }
+
+            var breadcrumbs = @event["breadcrumbs"] as Dictionary<string, object>[];
+            foreach (var crumb in breadcrumbs)
+            {
+                var crumbMetadata = crumb["metaData"] as Dictionary<string, object>;
+                TruncateStringsInMetadataSection(crumbMetadata);
+            }
+
+            return SerializePayload(payload);
+        }
+
+        private void TruncateStringsInMetadataSection(Dictionary<string, object> section)
```

Comment:
The method is called recursively for dictionaries, not just metadata sections. How about:

```suggestion
        private void TruncateStringsInDictionary(Dictionary<string, object> section)
```

Reply:
Done 👍